### PR TITLE
Keeping actions up to date automatically

### PR DESCRIPTION
**Pull request by tolzhabayev**
- moving to pin-by-tag instead. Most of the actions are coming from github anyway.
- adding dependabot action to keep the actions up to date

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -107,7 +107,7 @@ jobs:
         shell: bash
 
       - name: Checkout specified branch
-        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        uses: actions/checkout@v4.2.2 # v4.2.2
```
Comment (xnyo) [inline .github/workflows/cd.yml:110]: Do we have to keep the comments or can we just change them to:

```suggestion
        uses: actions/checkout@v4.2.2
```

Comment (tolzhabayev) [reply .github/workflows/cd.yml:110]: true 
vs35fh1ruef51

Comment (tolzhabayev) [reply .github/workflows/cd.yml]: done

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Set update schedule for GitHub Actions
```
Comment (tolzhabayev) [inline .github/dependabot.yml:1]: I am not sure if `directory: /` would also work for https://github.com/grafana/plugin-ci-workflows/blob/814f763609866edfb5df79d7e9a1eed7609bb210/actions/plugins/setup/action.yml#L19

but I guess we just have to try it out and see

**Review by xnyo — APPROVED** [review]
LGTM!

**Pull request merged**